src/AppInstallerRepositoryCore/Microsoft/SQLiteIndex.h

```diff
@@ -146,6 +146,12 @@ namespace AppInstaller::Repository::Microsoft
         // Largely a utility function; should not be used to do work on behalf of the index by the caller.
         Utility::NormalizedName NormalizeName(std::string_view name, std::string_view publisher) const;
 
+        // Get all the dependencies for a specific manifest.
+        std::map<Manifest::Dependency, SQLite::rowid_t> GetDependenciesByManifestRowId(SQLite::rowid_t manifestRowId) const;

```

Comment:
```suggestion
        std::vector<std::pair<IdType, Utility::Version>> GetDependenciesByManifestId(IdType manifestId) const;
```
We shouldn't build the dependency internally; that can be done with the existing interface.

Reply:
Why is this a `map`?
Please use `IdType`.
Why are you returning a second row id?  What is it?  The comment should indicate what the ids are that you are returning.